qutip/solve/piqs.py

```diff
@@ -98,6 +98,18 @@
     "Pim",
 ]
 
+
+def _ensure_int(x):
+    """
+    Ensure that a floating-point value `x` is exactly an integer, and return it
+    as an int.
+    """
+    out = int(x)
+    if out != x:
+        raise ValueError(f"{x} is not an integral value")
+    return out
```

Comment:
Not a fan, there should be a tolerance when comparing an int and float. 
And I think the test should be in the function calling this, not here, this will make the error harder to trace.

Reply:
Ok, let's just leave keep in mind that piqs will need some attention eventually.